src/pdf_beamtime/src/pdf_beamtime_server.cpp

```diff
@@ -51,13 +52,32 @@ PdfBeamtimeServer::PdfBeamtimeServer(
   current_state_ = State::HOME;
   gripper_present_ = node_->get_parameter("gripper_present").as_bool();
   inner_state_machine_ = new InnerStateMachine(node_);
+
+  bluesky_interrupt_service_ = interrupt_node_->create_service<BlueskyInterruptMsg>(
+    "bluesky_interrupt",
+    std::bind(
+      &PdfBeamtimeServer::bluesky_interrupt_cb, this, _1, _2));
+}
+
+void PdfBeamtimeServer::bluesky_interrupt_cb(
```

Comment:
Some logic here couldn't hurt. Does the structure of the Service Message even matter? The response isn't even generated...